### DISCUSSION TITLE
Default Component Styling Based on (Un)supported / (Un)available

### DISCUSSION THREAD

**Discussion by cjpillsbury** [muxinc/media-chrome#135] [Proposals]
Depends on #134 

### Questions to answer

- Should we have any default styling for components based on states being (un)supported and/or (un)available?
  - NOTE: We will need these styles for the Open Elements effort regardless.
- If so, what should the defaults be?
  - "Hidden" styles if unsupported? "disabled" styles if unavailable?
- How should we handle "unknown" states? No default styles? Will the onus be on implementors to differentiate between unknown vs. supported/unsupported (depends on decisions from #134 )

**Comment by heff** [comment]
On hidden or not, I think that can't be built into the component itself. I think that needs to be a decision made by the larger context (e.g. control-bar or media-container), while making it really easy to hide if that's desired (which #134 does IMO). 

If we do support "unavailable" and have a default disabled style for buttons in general, I see no reason those shouldn't line up.

Unknown feels like a case by case thing. You're either gonna hide something until you know it's supported (airplay), or show it until you know it's not supported (volume). How often are those things actually async in a user-impactful way? Is there an example of something that will be unknown until the API is called and it either works or throws an error?

**Reply by cjpillsbury** [reply]
I think there's a 3rd option, which is effectively (part of) what we're making available in Mux Player based on a [stream.new](https://github.com/muxinc/stream.new) use case: we're aggregating all the async initial load stuff and stream.new will not show the chrome at all until all relevant initialization crud is loaded/known. Aka don't show _*any*_ of the chrome until we know e.g. volume is supported or unsupported.

**Reply by cjpillsbury** [reply]
> On hidden or not, I think that can't be built into the component itself. I think that needs to be a decision made by the larger context (e.g. control-bar or media-container), while making it really easy to hide if that's desired (which #134 does IMO).

for sure. I buy that.

**Reply by cjpillsbury** [reply]
> How often are those things actually async in a user-impactful way? Is there an example of something that will be unknown until the API is called and it either works or throws an error?

See previous convos in #134 for async examples of volume checking and airplay support. CC availability will also very frequently be unknown until src has fully loaded.

**Comment by gkatsev** [comment]
Related to default styles, default behavior is something we should consider. When I was adding #311 before making it consistent with the other unavailable attributes we have, I had the button disable itself, so it isn't clickable if it is showing. I also updated to null-check the behavior in media-controller, but it might be better for the buttons to become disabled in these cases so that they don't trigger an event in the first place. At least, I think we should call this out explicitly rather than just let it happen organically.
